cdap-master-spi/src/main/java/io/cdap/cdap/master/spi/twill/StatefulTwillPreparer.java

```diff
@@ -23,15 +23,16 @@
  * Extension interface for {@link TwillPreparer} to implement if it supports stateful execution of
  * {@link TwillRunnable}.
  */
-public interface StatefulTwillPreparer extends TwillPreparer {
+public interface StatefulTwillPreparer extends DependentTwillPreparer {
 
   /**
    * Declares the given runnable with stateful execution.
    *
-   * @param runnableName name of the {@link TwillRunnable}
+   * @param mainRunnableName name of the {@link TwillRunnable}
    * @param orderedStart {@code true} to start replicas one by one; {@code false} to start replicas in parallel
    * @param statefulDisk an optional list of {@link StatefulDisk} available for the runnable container
    * @return this {@link TwillPreparer}
    */
-  StatefulTwillPreparer withStatefulRunnable(String runnableName, boolean orderedStart, StatefulDisk... statefulDisk);
+  StatefulTwillPreparer withStatefulRunnable(String mainRunnableName, boolean orderedStart,
```

Comment:
From the stateful perspective, there is nothing called `main` runnable. We shouldn't need to change this.